js/components/addressBook/addressBook_controller.js

```diff
@@ -178,4 +181,9 @@ angular.module('contactsApp')
 		});
 	};
 
+	ctrl.toggleState = function() {
+		ctrl.enabled = AddressBookService.toggleState(ctrl.addressBook);
+		ContactService.fillCache();
```

Comment:
Why do you call `fillCache`?

Reply:
It was to force a full refresh of the cache. This pr is still wip, I'm not convinced yet! :)